### DISCUSSION TITLE
How to pass non tensor args to loaded JIT model?

### DISCUSSION THREAD

**Discussion by ostegm** [sugarme/gotch#130] [Q&A]
Hi there,

I'm trying to load the silero-vad JIT model into gotch. In python the model is used as follows:

```
model = torch.jit.load("silero_vad.jit", map_location=device)
model.eval()
audio_chunk = #raw bytes from mic
audio_int16 = np.frombuffer(audio_chunk, np.int16)
voiced_confidence = self.model(
        torch.from_numpy(audio_int16), 16000
    ).item()
```

Notice the model call takes a tensor and a sample rate value integer. When trying to do something similar in gotch, I get the following error:

```
}]2024/01/24 11:41:40 Libtorch API Error: forward() Expected a value of type 'int' for argument 'sr' but instead found type 'Tensor'.
Position: 2
```

Code below:

```
func main() {
	flag.Parse()

	// Load the Python saved module.
	model, err := ts.ModuleLoad(modelPath)
	if err != nil {
		log.Fatal(err)
	}

	// For now just use a random tensor as audio input.
	xs := ts.MustRand([]int64{4000}, gotch.Float, gotch.CPU)
        // Sample rate input
	sr, err := ts.OfSlice([]int64{16000})
	if err != nil {
		log.Fatal(err)
	}

	out, err := model.ForwardTs([]*ts.Tensor{xs, sr})
	if err != nil {
		log.Fatal(err)
	}

	fmt.Printf("%i", out)

}

```

**Comment by sugarme** [comment]
@ostegm,

Please look up at closed issues or discussion. I think someone has asked. Also look at `gotch/example` folder for JIT examples. There are several ways of using JIT with gotch. Please let me know if you still have issues. 

Cheers,

**Comment by ostegm** [comment]
So far I have not found quite what I'm after... The examples are helpful for working with tensors, but do not cover this topic specifically and the only relevant conversation I found was [this one](https://github.com/sugarme/gotch/issues/20) which covered multiple *tensor* inputs. The tricky part for me is one of my inputs is expected to be a tensor and the other is an integer.

I'm unfamiliar with use of ivalues - is this something you'd recommend I dig into?

**Comment by sugarme** [comment]
@ostegm,

Look at Ivalue option in the example and issues answers as well.  Please provide URL to Silero source so that someone can help trying out. Cheers.

**Comment by ostegm** [comment]
Just gonna go another route. Thanks
